lib/streamlit/type_util.py

```diff
@@ -445,11 +454,12 @@ def convert_anything_to_df(df: Any) -> DataFrame:
     if is_type(df, "numpy.ndarray") and len(df.shape) == 0:
         return pd.DataFrame([])
 
-    if is_type(df, _SNOWPARK_DF_TYPE_STR) and not isinstance(df, list):
-        df = pd.DataFrame(df.take(MAX_UNEVALUATED_DF_ROWS))
-        if df.shape[0] == MAX_UNEVALUATED_DF_ROWS:
+    if is_type(df, _SNOWPARK_DF_TYPE_STR) or is_type(df, _SNOWPARK_TABLE_TYPE_STR):
+        df = pd.DataFrame(df.take(max_unevaluated_rows))
+        if df.shape[0] == max_unevaluated_rows:
             st.caption(
-                f"⚠️ Showing only 10k rows. Call `collect()` on the dataframe to show more."
+                f"⚠️ Showing only {string_util.simplify_number(max_unevaluated_rows)} rows. "
+                f"Call `collect()` on the dataframe to show more."
```

Comment:
nit: I think you can remove the `f` here in the second line (since there are no variables to replace.